src/main/java/seedu/address/logic/commands/DoneCommand.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.task.Task;
+
+/**
+ * Deletes a task identified using it's displayed index from the address book.
+ */
+public class DoneCommand extends Command {
+
+    public static final String COMMAND_WORD = "done";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": mark the task identified by the index number used in the displayed task list as done.\n"
+            + "Parameters: INDEX (one or more positive integer that is separate by a white space)\n"
+            + "Example: " + COMMAND_WORD + " 1 2 3";
+
+    public static final String MESSAGE_Done_TASK_SUCCESS = "Task: %1$s is done";
+
+    private final Index[] targetIndexes;
+
+    public DoneCommand(Index[] targetIndexes) {
```

Comment:
What does this return if the arg is empty @dearvae

Reply:
if it is empty the parser will return a exception with error message, it won't be turned into a DoneCommand.

Reply:
but your concern make sense! i will add     requireNonNull(targetIndexes);